.github/labeler.yml

```diff
@@ -1,3 +1,31 @@
+# add labels for code changes
+python:
+  - "**/*.py"
+R-code:
+  - "**/*.R"
+bash:
+  - "**/*.sh"
+  - "**/*.bash"
+  - "**/postBuild"
+
+# add labels for config changes
+configuration:
+  - "**/*.yml"
+  - "**/*.yaml"
+  - "**/*.json"
+  - "**/*.r"
+  - "**/Dockerfile"
```

Comment:
wondering if this should have a 'code'-based label, config, or both?

Reply:
My tendency would be to do what you're doing here and try to tag like things. That being said, I'm trying to imagine cases where I'd need to go to our issues/PR pages and filter by some of these. Perhaps its more for assessment and reporting purposes rather than development or support. I think it'd be worthwhile to add these since it does not require a lot of resources for labeler to run, and it doesn't take up head space to manage all of them.

In the above, `**/*.r` is actually R code and not configuration.

I think that `Dockerfile` should be `code` or `dockerfile` or `image`, but not `configuration`. The latter applies to files that are for hub configuration. Maybe it should be subsumed by a `**/image` match for an `image` label.

Reply:
Or maybe `singleuser` instead of `image` to avoid confusion with `images` below.

Reply:
> Or maybe `singleuser` instead of `image` to avoid confusion with `images` below.

yeah.  or `singleuser-image`...

i can do some cool stuff like:
```
singleuser-image:
  - "**/image/*"
  - "**/images/*"
  - "!**/*.png"
```
this will catch all image configs etc.  now i can explicitly ignore these same paths and add the `configuration` label on all other `y(a)ml` files:

```
configuration:
  - "!**/image/*"
  - "!**/images/*"
  - "**/*.yml"
  - "**/*.yaml"
  - "**/*.json"
```